Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -4119,7 +4119,7 @@ public boolean moveCards(Set<? extends Card> cards, Zone toZone,
     @Override
     public boolean moveCards(Set<? extends Card> cards, Zone toZone, Ability source, Game game, boolean tapped, boolean faceDown, boolean byOwner, List<UUID> appliedEffects) {
         if (cards.isEmpty()) {
-            return true;
+            return false;
```

Comment:
If you look at my arguments at the top of the MR you see why I changed this value. I do see however that in other methods, often a false returned. I think that is odd, because if nothing has to be done, that means if you don't do anything, then it's completed. Why would having nothing to do, be a fail?

The MirrorWorksTest is failing, but I don't see my changes causing this. CardsImpl::getCards can't find the 2nd Mox Diamond...

Reply:
Please check my arguments at the top of my MR, which explains my reasoning on returning false for null/empty values.

Reply:
Yes, it's correct xmage logic from your new code and from your explain. I mean it can be break changes for some cards that checks a move result. E.g. if your effect instruct "move cards, add life" then it must add life even on empty move. So you must check move usage and try to find compare operation -- that usage can be potentially broken on empty move.

Reply:
Thanks! :) 
Yes, some cards might break, but since none of the tests are failing (except for Mirrorworks), we don't know which cards are wrongly implemented that way... Those can be considered as false positives. It is a risk we have to consider.

Reply:
> E.g. if your effect instruct "move cards, add life" then it must add life even on empty move.

Yes, sometimes there are cards implemented like:

if(player.moveCards()){
   for example: player.gainLife(xxx)
}

and sometimes there are cards implemented like:

player.moveCards();
player.gainLife(xxx);

So it depends per card, maybe we can do a regex search on if(XX.moveCards(yy) statements?

Reply:
It seems I did not update all moveCards() calls (with the list argument), will take a look at that too

Reply:
There are some playerImpl.moveCards() calls that explicitly return the value, so those might be different now. But no tests is failing on that. And in other instances we have something like
`
playerImpl.moveCards(xxx)
return true;
`